geoh5py/objects/object_base.py

```diff
@@ -271,7 +285,7 @@ def add_data_map(self, data: ReferencedData, name: str, values: dict):
         )
         data_maps[name] = geom_data
         data.data_maps = data_maps
-        return geom_data
+        return data_type
```

Comment:
You are returning the type? I think we should still return the data itself, like `add_data` does.
Curious why we don't have a test that catches that.